Documentation/Contributors/DocumentationGuide/README.md

```diff
@@ -0,0 +1,491 @@
+ */
+Matrix4.fromColumnMajorArray = function(values, result) {
+    //>>includeStart('debug', pragmas.debug);
+    if (!defined(values)) {
+        throw new DeveloperError('values is required.');
+    }
+    //>>includeEnd('debug');
+
+    return Matrix4.clone(values, result);
+};
+```
+
+## Examples
+
+Developers almost always jump to an example before reading the doc.  Provide concise but instructive code examples with enough context whenever possible.
+
+Useful examples:
+```javascript
+/**
+ * ...
+ *
+ * @example
+ * var n = Cesium.Math.lerp(0.0, 2.0, 0.5); // returns 1.0
+ */
+CesiumMath.lerp = function(p, q, time) {
+    // ...
+```
+
+```javascript
+/**
+ * ...
+ *
+ * @example
+ * // Apply non-uniform scale to node LOD3sp
+ * var node = model.getNode('LOD3sp');
+ * node.matrix = Cesium.Matrix4.fromScale(new Cesium.Cartesian3(5.0, 1.0, 1.0), node.matrix);
+ */
+Model.prototype.getNode = function(name) {
+    // ...
+```
+
+Unnecessary example:
+```javascript
+/**
+ * ..
+ *
+ * @example
+ * var f = Cesium.Math.EPSILON1;
+ */
+CesiumMath.EPSILON1 = 0.1;
+```
+
+* Use the Cesium namespace (`Cesium.`) in examples.
+* Limit code in `@example` tags to 80 lines so it does not overflow.
+
+## References
+
+* Use `@see` sparingly to link to related classes, functions, and online resources., e.g.,
+```javascript
+/**
+ * A {@link TerrainProvider} that produces terrain geometry by tessellating height maps
+ * retrieved from an ArcGIS ImageServer.
+ *
+ * ...
+ *
+ * @see TerrainProvider
+ */
+function ArcGisImageServerTerrainProvider(options) {
+    // ...
+```
+
+generates
+
+![](ArcGisImageServerTerrainProvider.png)
+
+* Use `#` to reference an instance member (e.g., one that is assigned to the prototype); use `.` to access a static member, e.g.,
+```
+@see Class
+@see Class#instanceMember
+@see Class.staticMember
+```
+* Use `{@link className}` to link to another documented type.  This is not required for `@param` tags when the type is provided.
+* Use `<code> </code>` tags when referring to parameters or other variable names and values within a description.
+* Use `{@link URL|title}` to link to external sites.
+
+## Classes
+
+Define a class with `@alias` and `@constructor` tags on the constructor function, e.g.,
+
+```javascript
+/**
+ * A 3D Cartesian point.
+ *
+ * @alias Cartesian3
+ * @constructor
+ *
+ * ...
+ */
+function Cartesian3(x, y, z) {
+   // ...
+```
+
+## Properties and Constants
+
+* Use `@type` and `@default` (whenever possible) to document properties, e.g.,
+```javascript
+function Cartesian3(x, y) {
+    /**
+     * The X component.
+     *
+     * @type {Number}
+     * @default 0.0
+     */
+    this.x = defaultValue(x, 0.0);
+
+    // ...
+```
+* Use `@memberOf` when documenting property getter/setters, e.g.,
+```javascript
+defineProperties(Entity.prototype, {
+    /**
+     * Gets or sets whether this entity should be displayed. When set to true,
```

Comment:
We might want to mention documenting readonly properties starting with "Gets.. " and read/write properties as "Gets or sets". Both of these examples do it, but it's not explicitly mentioned.